{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merging HGDP and 1000 Genomes\n",
    "\n",
    "This script will walk you through the merging of the HGDP and 1000 Genomes genotypes.\n",
    "We will follow these steps:\n",
    "- Download HGDP files, and transform them into a plink file\n",
    "- Download 1000G files and keep only SNPs found in the HGDP files\n",
    "- Merge the HGDP and 1000G files\n",
    "\n",
    "## Needed packages\n",
    "\n",
    "To run this script you'll need [plink](https://www.cog-genomics.org/plink2), [vcftools](https://vcftools.github.io/index.html), [bcftools](https://samtools.github.io/bcftools/bcftools.html).\n",
    "All of these can be found in [bioconda](https://bioconda.github.io/)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Preliminaries\n",
    "\n",
    "First let's import modules and set up paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob, os, shutil, subprocess, csv, time\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/tomas/Documents/Repos/HGDP_1000G_Merge\n"
     ]
    }
   ],
   "source": [
    "projpath = os.path.realpath(\"..\")\n",
    "pathhgdp = os.path.join(projpath, \"DataBases\", \"HGDP\")\n",
    "path1000 = os.path.join(projpath, \"DataBases\", \"1000G\")\n",
    "print(projpath)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## HGDP\n",
    "\n",
    "The HGDP files (Stanford) were downloaded from [here](http://hagsc.org/hgdp/files.html), and the sample list file from [here](http://www.stanford.edu/group/rosenberglab/data/rosenberg2006ahg/SampleInformation.txt).\n",
    "The script to transform the HGDP data to plink format is called HGDPtoPlink.sh and was modified from [here](http://www.harappadna.org/2011/02/hgdp-to-ped-conversion/).\n",
    "The HGDP data uses coordinates from build 36.1 (a list of assemblies can be found [here](https://genome.ucsc.edu/FAQ/FAQreleases.html))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "dos2unix: converting file HGDP_FinalReport_Forward.txt to Unix format...\n",
      "dos2unix: converting file HGDP_Map.txt to Unix format...\n",
      "dos2unix: converting file SampleInformation.txt to Unix format...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.21 64-bit (19 Oct 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to hgdp.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --missing-genotype -\n",
      "  --out hgdp\n",
      "  --output-missing-genotype 0\n",
      "  --tfile hgdp\n",
      "\n",
      "32108 MB RAM detected; reserving 16054 MB for main workspace.\n",
      "Processing .tped file... done.2131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899100%\n",
      "hgdp-temporary.bed + hgdp-temporary.bim + hgdp-temporary.fam written.\n",
      "660918 variants loaded from .bim file.\n",
      "1043 people (673 males, 370 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1043 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999003.\n",
      "660918 variants and 1043 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to hgdp.bed + hgdp.bim + hgdp.fam ... 10%"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: 1057 het. haploid genotypes present (see hgdp.hh ); many commands\n",
      "treat these as missing.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\b1112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.21 64-bit (19 Oct 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to hgdp940.log.\n",
      "Options in effect:\n",
      "  --bfile hgdp\n",
      "  --keep Sample_keep.txt\n",
      "  --make-bed\n",
      "  --out hgdp940\n",
      "\n",
      "32108 MB RAM detected; reserving 16054 MB for main workspace.\n",
      "660918 variants loaded from .bim file.\n",
      "1043 people (673 males, 370 females) loaded from .fam.\n",
      "--keep: 940 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 940 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.998995.\n",
      "660918 variants and 940 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to hgdp940.bed + hgdp940.bim + hgdp940.fam ... 6%"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: 971 het. haploid genotypes present (see hgdp940.hh ); many commands\n",
      "treat these as missing.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.21 64-bit (19 Oct 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to hgdp940.log.\n",
      "Options in effect:\n",
      "  --bfile hgdp940\n",
      "  --out hgdp940\n",
      "  --recode\n",
      "\n",
      "32108 MB RAM detected; reserving 16054 MB for main workspace.\n",
      "660918 variants loaded from .bim file.\n",
      "940 people (618 males, 322 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 940 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: 971 het. haploid genotypes present (see hgdp940.hh ); many commands\n",
      "treat these as missing.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total genotyping rate is 0.998995.\n",
      "660918 variants and 940 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to hgdp940.ped + hgdp940.map ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['bash', 'HGDPtoPlink.sh'], returncode=0)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Run the HGDPtoPlink script\n",
    "#It can take a while\n",
    "os.chdir(os.path.join(projpath, \"Code\"))\n",
    "subprocess.run([\"bash\", \"HGDPtoPlink.sh\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Because the 1000G uses the GRCh37 assembly (fasta file can be found [here](http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/) as human_g1k_v37.fasta.gz) we'll need to liftover the HGDP coordinates.\n",
    "To do that we'll use UCSC [liftOver](http://genome.ucsc.edu/cgi-bin/hgLiftOver) already installed using bioconda, and [liftOverPlink](https://github.com/sritchie73/liftOverPlink) as a wrapper to work with plink files (`ped` and `map` formats).\n",
    "The chain file that tells liftOver how to convert between hg18 and hg19 can be downloaded [here](http://hgdownload.cse.ucsc.edu/goldenPath/hg18/liftOver/hg18ToHg19.over.chain.gz)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Converting MAP file to UCSC BED file...\n",
      "SUCC:  map->bed succ\n",
      "Lifting BED file...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading liftover chains\n",
      "Mapping coordinates\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUCC:  liftBed succ\n",
      "Converting lifted BED file back to MAP...\n",
      "SUCC:  bed->map succ\n",
      "cleaning up BED files...\n",
      "PLINK v1.90b6.21 64-bit (19 Oct 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to lifted.log.\n",
      "Options in effect:\n",
      "  --extract snplist.txt\n",
      "  --file hgdp940\n",
      "  --out lifted\n",
      "  --recode\n",
      "\n",
      "32108 MB RAM detected; reserving 16054 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).11111111111111212121212121212121213131313131313131314141414141414141415151515151515151515161616161616161616171717171717171717171818181818181818181919191919191919192020202020202020202021212121212121212122222222222222222222232323232323232323242424242424242424252525252525252525252626262626262626262727272727272727272728282828282828282829292929292929292930303030303030303030313131313131313131323232323232323232323333333333333333333434343434343434343535353535353535353536363636363636363637373737373737373737383838383838383838393939393939393939404040404040404040404141414141414141414242424242424242424243434343434343434344444444444444444445454545454545454545464646464646464646474747474747474747474848484848484848484949494949494949495050505050505050505051515151515151515152525252525252525252535353535353535353545454545454545454555555555555555555555656565656565656565757575757575757575758585858585858585859595959595959595960606060606060606060616161616161616161626262626262626262626363636363636363636464646464646464646565656565656565656566666666666666666667676767676767676767686868686868686868696969696969696969707070707070707070707171717171717171717272727272727272727273737373737373737374747474747474747475757575757575757575767676767676767676777777777777777777777878787878787878787979797979797979798080808080808080808081818181818181818182828282828282828282838383838383838383848484848484848484858585858585858585858686868686868686868787878787878787878788888888888888888889898989898989898990909090909090909090919191919191919191929292929292929292929393939393939393939494949494949494949595959595959595959596969696969696969697979797979797979797989898989898989898999999999999999999100%\n",
      "Performing single-pass .bed write (660918 variants, 940 people).\n",
      "--file: lifted-temporary.bed + lifted-temporary.bim + lifted-temporary.fam474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394%\n",
      "written.\n",
      "660918 variants loaded from .bim file.\n",
      "940 people (618 males, 322 females) loaded from .fam.\n",
      "--extract: 643948 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 940 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999017.\n",
      "643948 variants and 940 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode ped to lifted.ped + lifted.map ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.21 64-bit (19 Oct 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to hgdp940hg19.log.\n",
      "Options in effect:\n",
      "  --file\n",
      "  --make-bed\n",
      "  --map good_lifted.map\n",
      "  --out hgdp940hg19\n",
      "  --ped lifted.ped\n",
      "\n",
      "32108 MB RAM detected; reserving 16054 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).11111111111111212121212121212121213131313131313131314141414141414141415151515151515151515161616161616161616171717171717171717171818181818181818181919191919191919192020202020202020202021212121212121212122222222222222222222232323232323232323242424242424242424252525252525252525252626262626262626262727272727272727272728282828282828282829292929292929292930303030303030303030313131313131313131323232323232323232323333333333333333333434343434343434343535353535353535353536363636363636363637373737373737373737383838383838383838393939393939393939404040404040404040404141414141414141414242424242424242424243434343434343434344444444444444444445454545454545454545464646464646464646474747474747474747474848484848484848484949494949494949495050505050505050505051515151515151515152525252525252525252535353535353535353545454545454545454555555555555555555555656565656565656565757575757575757575758585858585858585859595959595959595960606060606060606060616161616161616161626262626262626262626363636363636363636464646464646464646565656565656565656566666666666666666667676767676767676767686868686868686868696969696969696969707070707070707070707171717171717171717272727272727272727273737373737373737374747474747474747475757575757575757575767676767676767676777777777777777777777878787878787878787979797979797979798080808080808080808081818181818181818182828282828282828282838383838383838383848484848484848484858585858585858585858686868686868686868787878787878787878788888888888888888889898989898989898990909090909090909090919191919191919191929292929292929292929393939393939393939494949494949494949595959595959595959596969696969696969697979797979797979797989898989898989898999999999999999999100%\n",
      "Performing single-pass .bed write (643948 variants, 940 people).\n",
      "--file: hgdp940hg19-temporary.bed + hgdp940hg19-temporary.bim +14243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394%\n",
      "hgdp940hg19-temporary.fam written.\n",
      "643948 variants loaded from .bim file.\n",
      "940 people (618 males, 322 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 940 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999017.\n",
      "643948 variants and 940 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to hgdp940hg19.bed + hgdp940hg19.bim + hgdp940hg19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "os.chdir(pathhgdp)\n",
    "#Using liftover\n",
    "liftoverplink = os.path.join(projpath, \"Code\", \"liftOverPlink\")\n",
    "%run $liftoverplink --map hgdp940.map --out lifted --chain hg18ToHg19.over.chain.gz\n",
    "badlifts = os.path.join(projpath, \"Code\", \"rmBadLifts\")\n",
    "%run $badlifts --map lifted.map --out good_lifted.map --log bad_lifted.dat\n",
    "#Creating a list of snps to include in lifted version\n",
    "snps = pd.read_csv(\"good_lifted.map\", sep = \"\\t\", header = None)\n",
    "snps.iloc[:,1].to_csv(\"snplist.txt\", index = False)\n",
    "#Excluding snps and creating binary file\n",
    "subprocess.run([\"plink\", \"--file\", \"hgdp940\", \"--recode\", \"--out\", \"lifted\", \"--extract\", \"snplist.txt\" ])\n",
    "subprocess.run([\"plink\", \"--file\", \"--ped\", \"lifted.ped\", \"--map\", \"good_lifted.map\", \"--make-bed\", \"--out\", \"hgdp940hg19\"])\n",
    "\n",
    "#Removing some files\n",
    "for file in glob.glob(\"*.ped\"):\n",
    "    os.remove(file)\n",
    "    \n",
    "for file in glob.glob(\"*.map\"):\n",
    "    os.remove(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create a snplist.bed with chr and position information\n",
    "os.chdir(pathhgdp)\n",
    "with open(\"snplist.bed\", \"w\") as f:\n",
    "    out = subprocess.run([\"awk\", \"NR==FNR {h[$1] = 1; next} {if(h[$4]==1) print$0}\", \"snplist.txt\", \"snp151Common.bed\"],\n",
    "                         stdout=f, text=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Performing single-pass .bed write (643948 variants, 940 people).\n",
      "643948 variants loaded from .bim file.\n",
      "643948 variants and 940 people pass filters and QC.\n",
      "Finished file... \n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Read hgdp940hg19 log file\n",
    "with open(\"hgdp940hg19.log\") as myfile:\n",
    "    for num, line in enumerate(myfile, 1):\n",
    "        if \"variants\" in line:\n",
    "            print(line, end='')\n",
    "    print(\"Finished file... \\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "dest_dir = os.path.join(projpath, \"Results\")\n",
    "for filename in glob.glob(\"hgdp940hg19.*\"):\n",
    "    shutil.copy(filename, dest_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1000G\n",
    "\n",
    "The 1000G Phase 3 files were downloaded from [here](ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/).\n",
    "First, we will extract the list of snps (`snplist.txt`) from the HGDP dataset for each chromosome of the 1000G samples using vcftools.\n",
    "Then we will concatenate the different autosomal chromosomes in one file and convert it into a plink binary file using bcftools and plink."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr21.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr21_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 9620 out of a possible 1105538 Sites\n",
      "Run Time = 65.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr17.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr17_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 16561 out of a possible 2329288 Sites\n",
      "Run Time = 193.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr11.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr11_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 31920 out of a possible 4045628 Sites\n",
      "Run Time = 544.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr22.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr22_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 9507 out of a possible 1103547 Sites\n",
      "Run Time = 65.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr18.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr18_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 20119 out of a possible 2267185 Sites\n",
      "Run Time = 213.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr3.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr3_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 44503 out of a possible 5832276 Sites\n",
      "Run Time = 1038.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr20.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr20_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 16844 out of a possible 1812841 Sites\n",
      "Run Time = 154.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr12.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr12_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 31773 out of a possible 3868428 Sites\n",
      "Run Time = 514.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr15.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr15_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 19566 out of a possible 2424689 Sites\n",
      "Run Time = 222.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr14.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr14_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 21453 out of a possible 2655067 Sites\n",
      "Run Time = 261.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr1.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr1_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 49479 out of a possible 6468094 Sites\n",
      "Run Time = 1222.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr19.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr19_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 10694 out of a possible 1832506 Sites\n",
      "Run Time = 109.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr10.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr10_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 34428 out of a possible 3992219 Sites\n",
      "Run Time = 577.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr16.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr16_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 19680 out of a possible 2697949 Sites\n",
      "Run Time = 243.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr13.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr13_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 25010 out of a possible 2857916 Sites\n",
      "Run Time = 311.00 seconds\n",
      "\n",
      "VCFtools - 0.1.16\n",
      "(C) Adam Auton and Anthony Marcketta 2009\n",
      "\n",
      "Parameters as interpreted:\n",
      "\t--gzvcf ALL.chr2.phase3_shapeit2_mvncall_integrated_v5b.20130502.genotypes.vcf.gz\n",
      "\t--out chr2_extracted\n",
      "\t--recode\n",
      "\t--bed snplist.bed\n",
      "\n",
      "Using zlib version: 1.2.11\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=MEINFO,Number=4,Type=String,Description=\"Mobile element info of the form NAME,START,END<POLARITY; If there is only 5' OR 3' support for this call, will be NULL NULL for START and END\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SVLEN,Number=.,Type=Integer,Description=\"SV length. It is only calculated for structural variation MEIs. For other types of SVs; one may calculate the SV length by INFO:END-START+1, or by finding the difference between lengthes of REF and ALT alleles\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=TSD,Number=1,Type=String,Description=\"Precise Target Site Duplication for bases, if unknown, value will be NULL\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=EUR_AF,Number=A,Type=Float,Description=\"Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AFR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AMR_AF,Number=A,Type=Float,Description=\"Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=SAS_AF,Number=A,Type=Float,Description=\"Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=DP,Number=1,Type=Integer,Description=\"Total read depth; only low coverage data were counted towards the DP, exome data were not used\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "Warning: Expected at least 2 parts in INFO entry: ID=AA,Number=1,Type=String,Description=\"Ancestral Allele. Format: AA|REF|ALT|IndelType. AA: Ancestral allele, REF:Reference Allele, ALT:Alternate Allele, IndelType:Type of Indel (REF, ALT and IndelType are only defined for indels)\">\n",
      "After filtering, kept 2504 out of 2504 Individuals\n",
      "Outputting VCF file...\n",
      "\tRead 652201 BED file entries.\n",
      "After filtering, kept 53645 out of a possible 7081600 Sites\n",
      "Run Time = 1452.00 seconds\n"
     ]
    }
   ],
   "source": [
    "#Extracting snps found in the HGDP to the 1000G files\n",
    "#Takes time\n",
    "os.chdir(path1000)\n",
    "snplists = [\"snplist.txt\",\n",
    "            \"snplist.bed\"]\n",
    "for snpl in snplists:\n",
    "    shutil.copy(os.path.join(pathhgdp, snpl), path1000)\n",
    "for file in glob.glob(\"*chr[0-9]*.gz\"):\n",
    "    outname = file.split(\".\")[1] + \"_extracted\"\n",
    "    subprocess.run([\"vcftools\", \"--gzvcf\", file, \"--bed\", \"snplist.bed\", \"--recode\", \"--out\", outname])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Checking the headers and starting positions of 16 files\n",
      "Concatenating chr19_extracted.recode.vcf\t1.328019 seconds\n",
      "Concatenating chr22_extracted.recode.vcf\t1.180856 seconds\n",
      "Concatenating chr1_extracted.recode.vcf\t5.801693 seconds\n",
      "Concatenating chr16_extracted.recode.vcf\t2.359901 seconds\n",
      "Concatenating chr18_extracted.recode.vcf\t2.431561 seconds\n",
      "Concatenating chr20_extracted.recode.vcf\t2.013835 seconds\n",
      "Concatenating chr13_extracted.recode.vcf\t2.970436 seconds\n",
      "Concatenating chr2_extracted.recode.vcf\t6.499817 seconds\n",
      "Concatenating chr15_extracted.recode.vcf\t2.397736 seconds\n",
      "Concatenating chr14_extracted.recode.vcf\t2.564049 seconds\n",
      "Concatenating chr3_extracted.recode.vcf\t5.352377 seconds\n",
      "Concatenating chr11_extracted.recode.vcf\t4.042636 seconds\n",
      "Concatenating chr17_extracted.recode.vcf\t2.128680 seconds\n",
      "Concatenating chr10_extracted.recode.vcf\t4.118487 seconds\n",
      "Concatenating chr21_extracted.recode.vcf\t1.264438 seconds\n",
      "Concatenating chr12_extracted.recode.vcf\t3.814989 seconds\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CompletedProcess(args=['bcftools', 'concat', '-o', '1000g.vcf.gz', '-Oz', 'chr19_extracted.recode.vcf', 'chr22_extracted.recode.vcf', 'chr1_extracted.recode.vcf', 'chr16_extracted.recode.vcf', 'chr18_extracted.recode.vcf', 'chr20_extracted.recode.vcf', 'chr13_extracted.recode.vcf', 'chr2_extracted.recode.vcf', 'chr15_extracted.recode.vcf', 'chr14_extracted.recode.vcf', 'chr3_extracted.recode.vcf', 'chr11_extracted.recode.vcf', 'chr17_extracted.recode.vcf', 'chr10_extracted.recode.vcf', 'chr21_extracted.recode.vcf', 'chr12_extracted.recode.vcf'], returncode=0)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatfiles = glob.glob(\"chr[0-9]*.recode.vcf\")\n",
    "function = [\"bcftools\", \"concat\", \"-o\", \"1000g.vcf.gz\", \"-Oz\"]\n",
    "function.extend(concatfiles)\n",
    "subprocess.run(function)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Convert to binary plink\n",
    "subprocess.run([\"plink\", \"--vcf\", \"1000g.vcf.gz\", \"--make-bed\", \"--out\", \"1000Ghg19\" ])\n",
    "#Updating fam file\n",
    "allfam = pd.read_csv(\"integrated_call_samples_v2.20130502.ALL.ped\", header = None, skiprows = 1, sep = \"\\t\")\n",
    "oldfam = pd.read_csv(\"1000Ghg19.fam\", header = None, sep = \" \")\n",
    "updatedfam = pd.merge(oldfam, allfam, how = \"inner\", left_on = 1, right_on = 1)\n",
    "updatedfam.iloc[:,[6,1,7,8,9,5]].to_csv(\"1000Ghg19.fam\", sep = \" \", header = False, index = False)\n",
    "os.remove(\"1000g.vcf.gz\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in glob.glob(\"chr*.recode.vcf\"):\n",
    "    os.remove(file)\n",
    "    \n",
    "dest_dir = os.path.join(projpath, \"Results\")\n",
    "for filename in glob.glob(\"1000Ghg19.*\"):\n",
    "    shutil.copy(filename, dest_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge reference samples\n",
    "\n",
    "Now we will merge the 1000G and HGDP databases, both using the hg19 coordinates and with related people removed.\n",
    "We will attempt an initial merge, and then flip strands for the problematic SNPs. \n",
    "Then we will try a second attempt, and remove the still problematic snps from both databases, to then attempt the final merge."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b4 64-bit (20 Mar 2017)\n",
      "Options in effect:\n",
      "  --bfile hgdp940hg19_temp\n",
      "  --bmerge 1000Ghg19_flip_temp\n",
      "  --make-bed\n",
      "  --out hgdp1000ghg19\n",
      "\n",
      "Hostname: tomasgazelle\n",
      "Working directory: /home/tomas/Documents/HGDP_1000G_Merge/Results\n",
      "Start time: Thu May 17 09:36:00 2018\n",
      "\n",
      "Random number seed: 1526564160\n",
      "3865 MB RAM detected; reserving 1932 MB for main workspace.\n",
      "940 people loaded from hgdp940hg19_temp.fam.\n",
      "2504 people to be merged from 1000Ghg19_flip_temp.fam.\n",
      "Of these, 2504 are new, while 0 are present in the base dataset.\n",
      "Warning: Multiple positions seen for variant 'rs9460309'.\n",
      "Warning: Multiple positions seen for variant 'rs11967812'.\n",
      "Warning: Multiple positions seen for variant 'rs11975477'.\n",
      "Warning: Multiple positions seen for variant 'rs13233990'.\n",
      "Warning: Multiple positions seen for variant 'rs11771665'.\n",
      "Warning: Multiple positions seen for variant 'rs2249255'.\n",
      "Warning: Multiple positions seen for variant 'rs4935071'.\n",
      "Warning: Multiple positions seen for variant 'rs9507310'.\n",
      "Warning: Multiple positions seen for variant 'rs1475276'.\n",
      "Warning: Multiple positions seen for variant 'rs12433837'.\n",
      "Warning: Multiple positions seen for variant 'rs1017238'.\n",
      "Warning: Multiple positions seen for variant 'rs2362394'.\n",
      "Warning: Multiple positions seen for variant 'rs1495901'.\n",
      "Warning: Multiple positions seen for variant 'rs9624480'.\n",
      "644003 markers loaded from hgdp940hg19_temp.bim.\n",
      "640456 markers to be merged from 1000Ghg19_flip_temp.bim.\n",
      "Of these, 0 are new, while 640456 are present in the base dataset.\n",
      "Performing single-pass merge (3444 people, 644003 variants).\n",
      "Merged fileset written to hgdp1000ghg19-merge.bed + hgdp1000ghg19-merge.bim +\n",
      "hgdp1000ghg19-merge.fam .\n",
      "644003 variants loaded from .bim file.\n",
      "3444 people (1851 males, 1593 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3432 founders and 12 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate is 0.995654.\n",
      "644003 variants and 3444 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to hgdp1000ghg19.bed + hgdp1000ghg19.bim + hgdp1000ghg19.fam ...\n",
      "done.\n",
      "\n",
      "End time: Thu May 17 09:36:32 2018\n",
      "\n"
     ]
    }
   ],
   "source": [
    "os.chdir(os.path.join(projpath, \"Results\"))\n",
    "#Preliminary merge\n",
    "subprocess.run([\"plink\", \"--bfile\", \"1000Ghg19\", \"--bmerge\", \"hgdp940hg19\", \"--make-bed\", \"--out\", \"hgdp1000ghg19\"])\n",
    "#Flip snps\n",
    "subprocess.run([\"plink\", \"--bfile\", \"1000Ghg19\", \"--flip\", \"hgdp1000ghg19-merge.missnp\", \"--make-bed\", \"--out\", \"1000Ghg19_flip\"])\n",
    "subprocess.run([\"plink\", \"--bfile\", \"1000Ghg19_flip\", \"--bmerge\", \"hgdp940hg19\", \"--make-bed\", \"--out\", \"hgdp1000ghg19\"])\n",
    "#Removing snps\n",
    "for file in glob.glob(\"*.bed\"):\n",
    "    outname = file.split(\".\")[0] + \"_temp\"\n",
    "    subprocess.run([\"plink\", \"--bfile\", file.split(\".\")[0], \"--exclude\", \"hgdp1000ghg19-merge.missnp\", \"--make-bed\", \"--out\", outname])\n",
    "\n",
    "#Removing temp files\n",
    "subprocess.run([\"plink\", \"--bfile\", \"hgdp940hg19_temp\", \"--bmerge\", \"1000Ghg19_flip_temp\", \"--make-bed\", \"--out\", \"hgdp1000ghg19\"])\n",
    "for file in glob.glob(\"*_temp*\"):\n",
    "    os.remove(file)\n",
    "    \n",
    "for file in glob.glob(\"*_flip*\"):\n",
    "    os.remove(file)\n",
    "\n",
    "with open(\"hgdp1000ghg19.log\", 'r') as fin:\n",
    "    file_contents = fin.read()\n",
    "    print(file_contents)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
